src/function.rs

```diff
@@ -372,6 +375,30 @@ fn path_exists(evaluator: &Evaluator, path: &str) -> Result<String, String> {
   )
 }
 
+fn pick(_evaluator: &Evaluator, count: &str, alphabet: Option<&str>) -> Result<String, String> {
+  let alphabet = alphabet.unwrap_or("0123456789abcdef");
+  if alphabet.is_empty() {
+    return Err(String::from("no characters in alphabet"));
+  }
+  let mut alphabet_unique = HashSet::<char>::with_capacity(alphabet.len());
+  for c in alphabet.chars() {
+    if !alphabet_unique.insert(c) {
+      return Err(format!("alphabet contains repeated character {c}"));
+    }
+  }
+  let alphabet = alphabet_unique.drain().collect::<Vec<char>>();
+  let len: usize = match count.parse::<usize>() {
+    Ok(l) => l,
+    _ => return Err(format!("Could not parse `{count}` as a positive integer")),
+  };
+  let mut rng = thread_rng();
+  let mut r = String::with_capacity(len);
+  while r.len() < len {
+    r.push(*alphabet.choose(&mut rng).unwrap());
```

Comment:
NIT: you might want to consider checking out the `IteratorRandom` trait in `rand::seq`, I think it might have an example that could make this more concise:

https://docs.rs/rand/latest/rand/seq/trait.IteratorRandom.html#method.choose_multiple_fill

or

https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.choose_multiple

Fortunately/unfortunately part of Rust is defininely digging in docs (glad they're there!) and finding the 👨‍🍳 💋 APIs to use -- there usually is a nice one for most things people want to do (in well written crates).

Reply:
Oh sorry you're absolutely right -- it does it without replacement which is not what we want.